File: frontend/web/monorepo/libs/mateu/src/mateu/shared/apiClients/dtos/componentmetadata/Rule.ts
import {RuleAction} from "./RuleAction";
import {RuleResult} from "./RuleResult";
import { RuleFieldAttribute } from "@mateu/shared/apiClients/dtos/componentmetadata/RuleFieldAttribute.ts";

export default interface Rule {

    filter: string
    action: RuleAction
    fieldName: string
    fieldAttribute: RuleFieldAttribute
    value: unknown
    expression: string | undefined
    result: RuleResult
    actionId: string

}

File: backend/shared/uidl/src/main/java/io/mateu/uidl/annotations/Rule.java
package io.mateu.uidl.annotations;

import io.mateu.uidl.data.RuleAction;
import io.mateu.uidl.data.RuleFieldAttribute;
import io.mateu.uidl.data.RuleResult;
import java.lang.annotation.Repeatable;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

@Repeatable(Rules.class)
@Retention(RetentionPolicy.RUNTIME)
public @interface Rule {

  String filter();

  RuleAction action();

  String fieldName();

  RuleFieldAttribute fieldAttribute();

  String value();

  String expression();

  String actionId();

  RuleResult result();
}
